taskwarrior_syncall/scripts/tw_asana_sync.py

```diff
@@ -0,0 +1,226 @@
+        if "asana_task_gid" in app_config:
+            asana_task_gid = app_config["asana_task_gid"]
+
+    # combination manually specified ----------------------------------------------------------
+    else:
+        config_args = {
+            "asana_workspace_gid": asana_workspace_gid,
+            "tw_project": tw_project,
+            "tw_tags": tw_tags,
+        }
+        if asana_task_gid is not None:
+            config_args["asana_task_gid"] = asana_task_gid
+        inform_about_config = True
+        combination_name = cache_or_reuse_cached_combination(
+            config_args,
+            config_fname="tw_asana_configs",
+            custom_combination_savename=custom_combination_savename,
+        )
+
+    # at least one of tw_tags, tw_project should be set ---------------------------------------
+    if not do_list_asana_workspaces and not tw_tags and not tw_project:
+        raise RuntimeError(
+            "You have to provide at least one valid tag or a valid project ID to use for"
+            " the synchronization"
+        )
+
+    # announce configuration ------------------------------------------------------------------
+    announce_items = {
+        "TW Tags": tw_tags,
+        "TW Project": tw_project,
+        "Asana Workspace GID": asana_workspace_gid,
+    }
+    if asana_task_gid is not None:
+        announce_items["Asana Task GID"] = asana_task_gid
+    logger.info(
+        format_dict(
+            header="Configuration",
+            items=announce_items,
+            prefix="\n\n",
+            suffix="\n",
+        )
+    )
+
+    # find token to connect to asana ---------------------------------------------------------
+    token = os.environ.get("ASANA_PERSONAL_ACCESS_TOKEN")
+    if token is not None:
+        logger.debug(
+            "Reading the Asana Personal Access Token (PAT) from environment variable..."
+        )
+    else:
+        token = fetch_from_pass_manager(token_pass_path)
+
+    assert token
+
+    # initialize taskwarrior ------------------------------------------------------------------
+    tw_side = TaskWarriorSide(tags=tw_tags, project=tw_project)
+
+    # initialize asana -----------------------------------------------------------------------
+    # client is a bit too verbose by default.
+    client_verbosity = max(verbose - 1, 0)
+    client = asana.Client.access_token(token)
+    asana_disable = client.headers.get("Asana-Disable", "")
+    client.headers["Asana-Disable"] = ",".join(
+        [client.headers.get("Asana-Disable", ""), "new_user_task_lists"]
+    )
+    client.options["client_name"] = "taskwarrior_syncall"
+
+    if do_list_asana_workspaces:
+        list_asana_workspaces(client)
+        return 0
+
+    # If we want to, we can iterate over all workspaces,
+    # and get all the tasks in each workspace. For now,
+    # require user to specify the workspace to pull tasks
+    # from.
+    assert asana_workspace_gid or asana_task_gid
```

Comment:
Instead of asserting, would it be better to check and raise a `RuntimeError` with a descriptive error message?

Reply:
Yes :) Will do

Reply:
Done in latest commit